### DISCUSSION TITLE
Expected behavior from radio in active scan

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#8893] [Q&A]
Hi,

I'm wondering about expected behavior for radio driver in MAC active scan that support IEEE 802.15.4 std.(2015) as per this standard to receive beacons in active scan that doesn't have destination address and destination Pan ID then device should set macImplicitBroadcast PIB to true.

so active scan doesn't depend on just setting device Pan ID to 0xFFFF before active scan and this will lead device to filter packets that doesn't have destination Pan ID and destination address(beacons packets) if macImplicitBroadcast PIB is False (which is the default value).

How could radio driver that support 802.15. std.(2015) filter beacons in active scan?

Thanks in advance

**Comment by jwhui** [comment]
Primary purpose of Active Scan is to discover PAN ID conflicts. For this reason, MAC Active Scan for use with Thread should set `macImplicitBroadcast` to true.

**Reply by testter2** [reply]
How could from radio perspective to know that MAC Active scan is started from MAC layer so that radio driver sets macImplicitBroadcast to true?

why shouldn't macImplicitBroadcast PIB be set from MAC layer ?

**Reply by jwhui** [reply]
OpenThread does not use MAC Active scan for any other purpose. As a result, the OpenThread radio driver can safely set the `macImplicitBroadcast` PIB to true.

**Reply by testter2** [reply]
@jwhui 

Setting macImplicitBroadcast PIB to true in Active Scan provides a special handling in radio to parse all frames in Tx mode to know whether or not a beacon request is to be transmitted or not. This is for setting the PIB to True.

Expected that this flag shall be set to False again after Active scan is done (for not passing high rate of broadcasted packets to upper layer in case of high congestive networks) ... How could radio layer knows the end of Active scan activity?

Propose to add optional interface to set this flag with start of active scan and clear it with stop.

**Reply by testter2** [reply]
@jwhui any update ?
